### ISSUE TITLE
False positive: clojure.core/with-local-vars bindings

### ISSUE THREAD

**Issue by kommen** [borkdude/clj-kondo#655]
**version**
clj-kondo v2019.11.23

**platform**

native binary on macOS

**problem**

clj-kondo shoudn't warn about undefined vars in `clojure.core/with-local-vars`' `name-vals-vec` vector.

**repro**

```
 echo '(with-local-vars [n 0] n)' | clj-kondo --lint -   
<stdin>:1:19: error: unresolved symbol n
linting took 53ms, errors: 1, warnings: 0
```

**expected behavior**

No error.

I'd be happy to provide a PR myself. For now I put a `:lint-as {clojure.core/with-local-vars clojure.core/let}` in my local config. Would that be an acceptable addition to the default-config?

**Comment by borkdude** [comment]
I think it would good to add in analyzer.clj here:

https://github.com/borkdude/clj-kondo/blob/6e342b086ecb3f02dc1ed173c5eec2d6648793f2/src/clj_kondo/impl/analyzer.clj#L1062

**Comment by kommen** [comment]
@borkdude thanks for the pointer!
